### ISSUE TITLE
Compiler hangs (infinite loop), when transpile( ) receives invalid gate names 

### ISSUE THREAD

**Issue by MattePalte** [Qiskit/qiskit-terra#7636]
### Environment

- **Qiskit Terra version**: 0.19.1
- **Python version**: 3.8
- **Operating system**: Ubuntu 18.04.6 LTS

### What is happening?

Running the transpiler to translate a circuit (to an equivalent gate set) hangs the program and never terminates. This happens because the argument `basic_gates` can only accept certain forms of names, e.g., 'h' is accepted but not 'hgate' . And in case the user provides a different name or the configuration file of the backend contains a typo, then the program runs forever without clue (potentially stopping also a backend server in a malicious context).

### How can we reproduce the issue?

Run the following script:
```python
import qiskit
from qiskit import QuantumCircuit
from qiskit import Aer, transpile, execute

circ = QuantumCircuit(3)
circ.h(0)
circ.cx(0, 1)
circ.cx(0, 2)
circ.measure_all()

decomposed = transpile(circ,
                       basis_gates=['rxgate', 'rygate', 'rzgate', 'hgate', 'cxgate'])  
backend = Aer.get_backend('qasm_simulator')
count = execute(decomposed, backend=backend, shots=1024).result().get_counts(decomposed)
```

changing the basic gates to valid names works: 
```python
basis_gates=['rx', 'ry', 'rz', 'h', 'cx']
```

### What should happen?

I expect the transpiler to fail with an error if I give unsupported gate name. Since this current behavior could hang an entire compiler if the gate names in a backend configuration file are misspelled.

### Any suggestions?

The loop happens at this location in the loop of the `_basic_search` function: 
https://github.com/Qiskit/qiskit-terra/blob/fcec842f1de9fd12120e30a1bf73bf7c52b1bf81/qiskit/transpiler/passes/basis/basis_translator.py#L343

I think that the error is in the input validation here, there is only check that the `basic_gates` are strings and not that they are valid gate names. 

The buggy code is probably this:
https://github.com/Qiskit/qiskit-terra/blob/fcec842f1de9fd12120e30a1bf73bf7c52b1bf81/qiskit/compiler/transpiler.py#L699

```python
def _parse_basis_gates(basis_gates, backend, circuits):
    ...
    # basis_gates could be None, or a list of basis, e.g. ['u3', 'cx']
    if basis_gates is None or (
        isinstance(basis_gates, list) and all(isinstance(i, str) for i in basis_gates)
    ):
        basis_gates = [basis_gates] * len(circuits)
```
we should probably check that the string belongs to the set of all possible and valid gate names.

**Comment by TheGupta2012** [comment]
Can I have a go at this?

**Comment by jakelishman** [comment]
This is similar to #5539, perhaps a slightly different manifestation because here there are _no_ valid gates in the set.  It's not actually an infinite loop, it's just unacceptably slow - if you let it run for a couple of minutes, it will eventually report that the basis is unreachable.

The new basis-translation strategy in #7211 I would imagine (but haven't verified) will fix this directly and have the failure be very fast.  We'll immediately detect that we can't use the given gates to generate anything, because the strategy is changed to expand out from the target set.  I'll close this as a duplicate of #5539, and we'll try to get #7211 merged soon - the blocker on it was retworkx 0.11, but that was released a few weeks ago.
